polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala

```diff
@@ -57,12 +57,13 @@ class PolyKernel private[kernel] (
     */
   protected lazy val taskManager: TaskManager = TaskManager(statusUpdates).unsafeRunSync()
 
-  // TODO: duplicates logic from runCell
   private def runPredef(interp: LanguageInterpreter[IO], language: String): IO[Stream[IO, Result]] =
     interp.init() *> {
-      val index = notebookContext.first.map(_.id).getOrElse(0.toShort) - 1
+      // Since there can be multiple predefs (one per interpreter), they get monotonically decreasing negative cell IDs
+      // starting with -1. Get the smallest existing ID, make sure it's 0 or less, and subtract one.
```

Comment:
Just changed `index` to `cellId`, added some explanation of what's happening, and added `math.min` to ensure we're starting with `-1` (in case notebook has weird IDs that don't start at 0 for some reason)